.circleci/config.yml

```
@@ -570,6 +571,24 @@ jobs:
       - checkout
       - attach_workspace:
           at: .
+      - run:
+          name: 'Detect newly introduced tests'
+          command: |
+            testForDebugging="<< parameters.test >>"
+
+            if [ -z "$testForDebugging" ]; then
+              tests=$(git diff origin/main --name-status --oneline --diff-filter=A | (grep 'src/test/regress/sql/' || true) | rev | cut -d "/" -f1 | rev | sed 's/.sql//g' | tr '\n' ' ' | sed 's/.$//')
```

Slightly simplified and working if we ever change the name of the main branch again:
```suggestion
              tests=$(git diff origin/HEAD --name-only --diff-filter=A | (grep '^src/test/regress/sql/' || true) | rev | cut -d "/" -f1 | rev | sed 's/.sql//g' | tr '\n' ' ' | sed 's/.$//')
```